diff --git a/node_modules/discord-player/dist/Player.js b/node_modules/discord-player/dist/Player.js
index 5b7c224..5ce3472 100644
--- a/node_modules/discord-player/dist/Player.js
+++ b/node_modules/discord-player/dist/Player.js
@@ -321,7 +321,7 @@ class Player extends tiny_typed_emitter_1.TypedEmitter {
                     const spotifyTrack = new Track_1.default(this, {
                         title: spotifyData.name,
                         description: (_c = spotifyData.description) !== null && _c !== void 0 ? _c : "",
-                        author: (_e = (_d = spotifyData.artists[0]) === null || _d === void 0 ? void 0 : _d.name) !== null && _e !== void 0 ? _e : "Unknown Artist",
+                        author: spotifyData.artists?.map(x => x.name).join(", ") || "Unknown Artist", // ! patch original: (_e = (_d = spotifyData.artists[0]) === null || _d === void 0 ? void 0 : _d.name) !== null && _e !== void 0 ? _e : "Unknown Artist",
                         url: (_g = (_f = spotifyData.external_urls) === null || _f === void 0 ? void 0 : _f.spotify) !== null && _g !== void 0 ? _g : query,
                         thumbnail: ((_k = (_j = (_h = spotifyData.album) === null || _h === void 0 ? void 0 : _h.images[0]) === null || _j === void 0 ? void 0 : _j.url) !== null && _k !== void 0 ? _k : (_l = spotifyData.preview_url) === null || _l === void 0 ? void 0 : _l.length)
                             ? `https://i.scdn.co/image/${(_m = spotifyData.preview_url) === null || _m === void 0 ? void 0 : _m.split("?cid=")[1]}`
@@ -365,7 +365,7 @@ class Player extends tiny_typed_emitter_1.TypedEmitter {
                             const data = new Track_1.default(this, {
                                 title: (_a = m.name) !== null && _a !== void 0 ? _a : "",
                                 description: (_b = m.description) !== null && _b !== void 0 ? _b : "",
-                                author: (_d = (_c = m.artists[0]) === null || _c === void 0 ? void 0 : _c.name) !== null && _d !== void 0 ? _d : "Unknown Artist",
+                                author: m.artists?.map(x => x.name).join(", ") || "Unknown Artist", // ! patch original: (_d = (_c = m.artists[0]) === null || _c === void 0 ? void 0 : _c.name) !== null && _d !== void 0 ? _d : "Unknown Artist"
                                 url: (_f = (_e = m.external_urls) === null || _e === void 0 ? void 0 : _e.spotify) !== null && _f !== void 0 ? _f : query,
                                 thumbnail: (_h = (_g = spotifyPlaylist.images[0]) === null || _g === void 0 ? void 0 : _g.url) !== null && _h !== void 0 ? _h : "https://www.scdn.co/i/_global/twitter_card-default.jpg",
                                 duration: Util_1.Util.buildTimeCode(Util_1.Util.parseMS(m.duration_ms)),
@@ -384,7 +384,7 @@ class Player extends tiny_typed_emitter_1.TypedEmitter {
                             const data = new Track_1.default(this, {
                                 title: (_a = m.track.name) !== null && _a !== void 0 ? _a : "",
                                 description: (_b = m.track.description) !== null && _b !== void 0 ? _b : "",
-                                author: (_d = (_c = m.track.artists[0]) === null || _c === void 0 ? void 0 : _c.name) !== null && _d !== void 0 ? _d : "Unknown Artist",
+                                author: m.track.artists?.map(x => x.name).join(", ") || "Unknown Artist", // ! patch original: (_d = (_c = m.track.artists[0]) === null || _c === void 0 ? void 0 : _c.name) !== null && _d !== void 0 ? _d : "Unknown Artist",
                                 url: (_f = (_e = m.track.external_urls) === null || _e === void 0 ? void 0 : _e.spotify) !== null && _f !== void 0 ? _f : query,
                                 thumbnail: (_j = (_h = (_g = m.track.album) === null || _g === void 0 ? void 0 : _g.images[0]) === null || _h === void 0 ? void 0 : _h.url) !== null && _j !== void 0 ? _j : "https://www.scdn.co/i/_global/twitter_card-default.jpg",
                                 duration: Util_1.Util.buildTimeCode(Util_1.Util.parseMS(m.track.duration_ms)),
